Makefile

```diff
@@ -158,10 +158,10 @@ e2e-vault: install-driver
 # Update CRDs in Helm Chart
 release: manifests
 	cp config/crd/bases/* charts/secrets-store-csi-driver/templates
-	cp config/crd/bases/* deploy/
+	cp config/crd/bases/* manifest_staging/deploy/
```

Comment:
where do we use `make release`?

can these go under `manifests`?

do we also need `manifest_staging` for chart updates?

Reply:
> where do we use make release?

`make release` is used in `promote-staging-manifest`

> do we also need manifest_staging for chart updates?

For charts, we can always add `semver` compare checks while adding new flags. The latest master chart is always used to run e2e tests. If we do maintain a copy of those charts in the `manifest_staging` dir, then we would need to run tests against those. For deploy manifests we need `manifest_staging` because the manifest are bound to the version, but with helm the semver compare does the trick. WDYT?

Reply:
The rationale behind adding the latest changes to `manifest_staging` instead of master is to ensure all configuration changes are released at the same time as the image in case there's any compatibility issues. e.g. a new flag. If a user grabs the chart from master, it should just work. I see your point about `semver` compare, but would be nice to not having to do that every time we make a chart update. As for e2es, we should test against `manifest_staging` chart.

Reply:
Thank you! Makes sense. I'll add that.